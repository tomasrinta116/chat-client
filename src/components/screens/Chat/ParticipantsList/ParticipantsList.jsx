import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListSubheader from "@material-ui/core/ListSubheader";
import styles from './ParticipantsList.styles';
import PropTypes from 'prop-types';

const ParticipantsList = props => {
  const {
    classes: { participantsListContainer },
    isOpen,
    toggle,
    participants
  } = props;

  const formattedList = (
    <div className={participantsListContainer}>
      <List
        subheader={<ListSubheader color="primary">Active Users</ListSubheader>}
      >
        {participants.map((participant, index) => {
          return (
            <ListItem key={participant + index.toString()}>
              {participant}
            </ListItem>
          );
        })}
      </List>
    </div>
  );

  return (
    <React.Fragment>
      <Drawer open={isOpen} onClick={toggle}>
        <div>{formattedList}</div>
      </Drawer>
    </React.Fragment>
  );
};

ParticipantsList.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  participants: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(ParticipantsList);
