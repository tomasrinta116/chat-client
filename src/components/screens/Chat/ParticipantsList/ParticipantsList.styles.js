const styles = {
    participantsListContainer: {
        width: 250
      }
}

export default styles;