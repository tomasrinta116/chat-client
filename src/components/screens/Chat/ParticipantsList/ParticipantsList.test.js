import React from "react";
import { shallow } from "enzyme";
import ParticipantsList from "./ParticipantsList";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";

describe("<ParticipantsList />", () => {
  const props = {
    isOpen: true,
    toggle: jest.fn(),
    participants: ["One", "Two", "Three", "Four"]
  };

  test("it renders correctly", () => {
    const wrapper = shallow(<ParticipantsList {...props} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test("it renders the right number of items", () => {
    const wrapper = shallow(<ParticipantsList {...props} />).dive();
    const list = wrapper.find(List).dive();
    const listItem = list.find(ListItem);
    //console.log(list.dive().debug());
    expect(list).toHaveLength(1);
    expect(listItem).toHaveLength(4);
  });
});
