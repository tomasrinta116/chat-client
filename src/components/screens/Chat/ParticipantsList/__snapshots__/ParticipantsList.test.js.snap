// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantsList /> it renders correctly 1`] = `
<Fragment>
  <WithStyles(Drawer)
    onClick={[MockFunction]}
    open={true}
  >
    <div>
      <div
        className="ParticipantsList-participantsListContainer-1"
      >
        <WithStyles(List)
          subheader={
            <WithStyles(ListSubheader)
              color="primary"
            >
              Active Users
            </WithStyles(ListSubheader)>
          }
        >
          <WithStyles(ListItem)
            key="One0"
          >
            One
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            key="Two1"
          >
            Two
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            key="Three2"
          >
            Three
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            key="Four3"
          >
            Four
          </WithStyles(ListItem)>
        </WithStyles(List)>
      </div>
    </div>
  </WithStyles(Drawer)>
</Fragment>
`;
